ast/src/asm_analysis/mod.rs

```diff
@@ -835,7 +835,7 @@ impl AnalysisASMFile {
     pub fn get_machine(&self, ty: &AbsoluteSymbolPath) -> Option<&Machine> {
         let mut path = ty.clone();
         let name = path.pop().unwrap();
-        self.modules[&path].machines.get(&name)
+        self.modules.get(&path)?.machines.get(&name)
```

Comment:
I added this small change, which is not 100% related to the PR, but I don't think it's worth opening a new one just for this.